tests/addnode.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import e2e_args
+import infra.ccf
+import infra.proc
+import json
+
+import logging
+import time
+
+from loguru import logger as LOG
+
+
+def create_and_add_node(
+    network, args, node_id, member_host, member_port, should_succeed=True
+):
+    dict_args = vars(args)
+    forwarded_args = {
+        arg: dict_args[arg] for arg in infra.ccf.Network.node_args_to_forward
+    }
+    node_status = args.node_status or "pending"
+    new_node = network.create_node(node_id, "localhost")
+    new_node.start(
+        lib_name=args.package,
+        node_status=node_status,
+        workspace=args.workspace,
+        label=args.label,
+        **forwarded_args
+    )
+    new_node_info = new_node.remote.info()
+    new_cert_path = "{}/{}".format(
+        new_node.remote.remote.root, getattr(new_node.remote, "pem")
+    )
+
+    if should_succeed:
+        new_quote_path = "{}/{}".format(
+            new_node.remote.remote.root, getattr(new_node.remote, "quote")
+        )
+    else:
```

Comment:
I think this branch is a little misleading with regards to the name of the function (`create_and_add_node`). Instead, we could pass the "lib_name" to `create_and_add_node` and start with that package. In the case of `libloggingenc` creating and adding the node will succeed, otherwise (e.g. `libluagenericenc`), the command will return with `CODE_ID_NOT_FOUND`.